frontend/src/components/Login.js

```diff
@@ -0,0 +1,277 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components/macro';
+import { HeartFillIcon, MailIcon, KeyIcon } from '@primer/octicons-react';
+
+import { sign } from '../reducers/user';
+
+const Login = () => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [mode, setMode] = useState(null);
+
+  const accessToken = useSelector((store) => store.user.accessToken);
+  const errors = useSelector((store) => store.user.errors);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  useEffect(() => {
+    if (accessToken) {
+      history.push('/');
+    }
+  }, [accessToken, history]);
+
+  const onFormSubmit = (e) => {
+    e.preventDefault();
+    dispatch(sign(name, email, password, mode));
+  };
+
+  return (
+    <PageWrapper>
+      <h1>Welcome to SecretMessage!</h1>
+      <h2>Please log in or register as a user</h2>
+      <Form onSubmit={onFormSubmit}>
+        <Wrapper>
+          <Container>
+            <Heart size={24} />
+            <InputBox
+              id='name'
+              maxLength='20'
+              type='text'
+              value={name}
+              onChange={(e) => setName(e.target.value)}
+              placeholder='Name'
+            />
+            <Label htlFor='name'>Name</Label>
```

Comment:
❕❕ The correct usage of Label and for in react is: (I think this just may have been a spell error 😅)
```suggestion
            <Label htmlFor='name'>Name</Label>
```
This also applies to line `63`